veggieseasons/lib/main.dart

```diff
@@ -16,16 +16,74 @@ void main() {
     DeviceOrientation.portraitDown,
   ]);
 
-  runApp(
-    ScopedModel<AppState>(
-      model: AppState(),
+  // The root scope is usually injected automatically by the
+  // CupertinoApp/MaterialApp. However, in this particular app there are widgets
+  // above the CupertinoApp that need their state restored, so we are injecting
+  // the root scope manually above everything.
+  runApp(RestorationScope.root(
```

Comment:
this code isn't on master yet is it? Are there non-root constructors?

Reply:
The code currently lives in a personal branch of mine. I am in the process of cleaning that up and then will make it available.

There are other constructors. The default one just starts a new child-scope within its parent scope. The root one actually inserts the data received from the engine into the tree.